### ISSUE TITLE
Problem compilling flasher

### ISSUE THREAD

**Issue by Demonic098** [balrog-kun/lights#4]
I have tried to compile flasher but i gets many errors. Optiboot compiled normally. Here is log :

`
make flasher
avr-gcc -gstabs -DF_CPU=16000000 -I -Os -Wall -Wstrict-prototypes -std=gnu99     flasher.c   -o flasher
In file included from passthrough.c:7:0,
                 from flasher.c:9:
/usr/lib/avr/include/avr/io.h:623:6: warning: #warning "device type not defined" [-Wcpp]
 #    warning "device type not defined"
      ^
In file included from /usr/lib/avr/include/avr/io.h:99:0,
                 from passthrough.c:7,
                 from flasher.c:9:
/usr/lib/avr/include/avr/wdt.h: In function ‘wdt_enable’:
/usr/lib/avr/include/avr/wdt.h:406:21: error: ‘WDT’ undeclared (first use in this function)
  if (_SFR_IO_REG_P (_WD_CONTROL_REG))
                     ^
/usr/lib/avr/include/avr/wdt.h:406:21: note: each undeclared identifier is reported only once for each function it appears in
/usr/lib/avr/include/avr/wdt.h:417:24: error: ‘WDCE’ undeclared (first use in this function)
     "r" ((uint8_t)(_BV(_WD_CHANGE_BIT) | _BV(WDE))),
                        ^
/usr/lib/avr/include/avr/wdt.h:417:42: error: ‘WDE’ undeclared (first use in this function)
     "r" ((uint8_t)(_BV(_WD_CHANGE_BIT) | _BV(WDE))),
                                          ^
/usr/lib/avr/include/avr/wdt.h: In function ‘wdt_disable’:
/usr/lib/avr/include/avr/wdt.h:446:21: error: ‘WDT’ undeclared (first use in this function)
  if (_SFR_IO_REG_P (_WD_CONTROL_REG))
                     ^
/usr/lib/avr/include/avr/wdt.h:460:37: error: ‘WDCE’ undeclared (first use in this function)
     [WDCE_WDE]  "n"  ((uint8_t)(_BV(_WD_CHANGE_BIT) | _BV(WDE)))
                                     ^
/usr/lib/avr/include/avr/wdt.h:460:55: error: ‘WDE’ undeclared (first use in this function)
     [WDCE_WDE]  "n"  ((uint8_t)(_BV(_WD_CHANGE_BIT) | _BV(WDE)))
                                                       ^
In file included from passthrough.c:22:0,
                 from flasher.c:9:
spi.h: In function ‘spi_mode’:
spi.h:15:2: error: ‘SPCR’ undeclared (first use in this function)
  SPCR = 0;
  ^
spi.h:16:31: error: ‘SPE’ undeclared (first use in this function)
  SPCR = (mode & 0x7F) | (1 << SPE) | (1 << MSTR);
                               ^
spi.h:16:44: error: ‘MSTR’ undeclared (first use in this function)
  SPCR = (mode & 0x7F) | (1 << SPE) | (1 << MSTR);
                                            ^
spi.h:17:2: error: ‘SPSR’ undeclared (first use in this function)
  SPSR = 1 << SPI2X; /* double speed */
  ^
spi.h:17:14: error: ‘SPI2X’ undeclared (first use in this function)
  SPSR = 1 << SPI2X; /* double speed */
              ^
spi.h:19:9: error: ‘SPDR’ undeclared (first use in this function)
  (void) SPDR;
         ^
spi.h: In function ‘spi_init’:
spi.h:7:18: error: ‘DDRB’ undeclared (first use in this function)
 #define SPI_DDR  DDRB
                  ^
spi.h:24:2: note: in expansion of macro ‘SPI_DDR’
  SPI_DDR |= SCK_PIN | MOSI_PIN;
  ^
spi.h: In function ‘spi_transfer’:
spi.h:35:2: error: ‘SPDR’ undeclared (first use in this function)
  SPDR = value;
  ^
spi.h:36:21: error: ‘SPSR’ undeclared (first use in this function)
  while (cnt -- && !(SPSR & (1 << SPIF)));
                     ^
spi.h:36:34: error: ‘SPIF’ undeclared (first use in this function)
  while (cnt -- && !(SPSR & (1 << SPIF)));
                                  ^
In file included from flasher.c:9:0:
nrf24.h: In function ‘nrf24_csn’:
passthrough.c:18:18: error: ‘PORTB’ undeclared (first use in this function)
 #define CSN_PORT PORTB
                  ^
nrf24.h:10:3: note: in expansion of macro ‘CSN_PORT’
   CSN_PORT |= CSN_PIN;
   ^
nrf24.h: In function ‘nrf24_ce’:
passthrough.c:16:18: error: ‘PORTC’ undeclared (first use in this function)
 #define CE_PORT  PORTC
                  ^
nrf24.h:64:3: note: in expansion of macro ‘CE_PORT’
   CE_PORT |= CE_PIN;
   ^
nrf24.h: In function ‘nrf24_init’:
passthrough.c:15:17: error: ‘DDRC’ undeclared (first use in this function)
 #define CE_DDR  DDRC
                 ^
nrf24.h:140:2: note: in expansion of macro ‘CE_DDR’
  CE_DDR |= CE_PIN;
  ^
passthrough.c:17:18: error: ‘DDRB’ undeclared (first use in this function)
 #define CSN_DDR  DDRB
                  ^
nrf24.h:141:2: note: in expansion of macro ‘CSN_DDR’
  CSN_DDR |= CSN_PIN;
  ^
In file included from flasher.c:9:0:
passthrough.c: In function ‘eeprom_read’:
passthrough.c:45:9: error: ‘EECR’ undeclared (first use in this function)
  while (EECR & (1 << EEPE));
         ^
passthrough.c:45:22: error: ‘EEPE’ undeclared (first use in this function)
  while (EECR & (1 << EEPE));
                      ^
passthrough.c:47:2: error: ‘EEAR’ undeclared (first use in this function)
  EEAR = addr;
  ^
passthrough.c:48:15: error: ‘EERE’ undeclared (first use in this function)
  EECR |= 1 << EERE; /* Start eeprom read by writing EERE */
               ^
passthrough.c:50:9: error: ‘EEDR’ undeclared (first use in this function)
  return EEDR;
         ^
passthrough.c: In function ‘setup’:
passthrough.c:55:14: error: ‘MCUCR’ undeclared (first use in this function)
  uint8_t m = MCUCR;
              ^
In file included from passthrough.c:23:0,
                 from flasher.c:9:
flasher.c: At top level:
nrf24.h:217:16: warning: ‘nrf24_rx_new_data’ defined but not used [-Wunused-function]
 static uint8_t nrf24_rx_new_data(void) {
                ^
In file included from passthrough.c:22:0,
                 from flasher.c:9:
spi.h: In function ‘spi_transfer’:
spi.h:38:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
In file included from flasher.c:9:0:
passthrough.c: In function ‘eeprom_read’:
passthrough.c:51:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
<builtin>: recipe for target 'flasher' failed
make: *** [flasher] Error 1

`

I am using the  ubuntu's avr-gcc 

Thanks,
Dee

**Comment by benchuk** [comment]
found a solution? also having problems compile this...

**Comment by balrog-kun** [comment]
Sorry for slow reply, I'll close this as duplicate of #3
